components/sceneutil/nodecallback.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef SCENEUTIL_NODECALLBACK_H
+#define SCENEUTIL_NODECALLBACK_H
+
+#include <osg/Callback>
+
+namespace osg
+{
+    class Node;
+    class NodeVisitor;
+}
+
+namespace SceneUtil
+{
+
+template <class Derived, typename NodeType=osg::Node*, typename VisitorType=osg::NodeVisitor*>
```

Comment:
Why need to customize `NodeType`? I don't see where it's used.

Reply:
It will be used once we convert remaining `osg::NodeCallback` uses. For example, `GeomMorpherController` expects to use a `MorphGeometry` node type.